cmd/server/search.go

```diff
@@ -34,60 +35,60 @@ type searchResponse struct {
 }
 
 // ACHFindNameOnly finds ACH Participants by name only
-func (s *searcher) ACHFindNameOnly(participantName string) ([]*fed.ACHParticipant, error) {
+func (s *searcher) ACHFindNameOnly(limit int, participantName string) ([]*fed.ACHParticipant, error) {
 	s.RLock()
 	defer s.RUnlock()
-	fi, err := s.ACHDictionary.FinancialInstitutionSearch(participantName)
-	if err != nil {
-		return nil, err
-	}
-	return fi, nil
+	fi := s.ACHDictionary.FinancialInstitutionSearch(participantName)
+	out := achLimit(fi, limit)
+	return out, nil
 }
 
 // ACHFindRoutingNumberOnly finds ACH Participants by routing number only
-func (s *searcher) ACHFindRoutingNumberOnly(routingNumber string) ([]*fed.ACHParticipant, error) {
+func (s *searcher) ACHFindRoutingNumberOnly(limit int, routingNumber string) ([]*fed.ACHParticipant, error) {
 	s.RLock()
 	defer s.RUnlock()
 	fi, err := s.ACHDictionary.RoutingNumberSearch(routingNumber)
 	if err != nil {
 		return nil, err
 	}
-	return fi, nil
+	out := achLimit(fi, limit)
+	return out, nil
 }
 
 // ACHFindCityOnly finds ACH Participants by city only
-func (s *searcher) ACHFindCityOnly(city string) []*fed.ACHParticipant {
+func (s *searcher) ACHFindCityOnly(limit int, city string) []*fed.ACHParticipant {
 	s.RLock()
 	defer s.RUnlock()
 	fi := s.ACHDictionary.CityFilter(city)
-	return fi
+	out := achLimit(fi, limit)
```

Comment:
Is there a reason to not pass limit into `CityFilter` ?

Reply:
Just keeping it at server for now